### ISSUE TITLE
no-unused-vars: single-underscore `_` parameter never reported with default options (false negative vs ESLint; valid eslint-disable directives flagged as unused)

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25626]
## Summary

oxlint's native `no-unused-vars` silently treats a single-underscore (`_`) function parameter as an intentionally-ignored placeholder and never reports it, even with no `argsIgnorePattern` configured. ESLint's default behavior reports it. This is a false negative that also makes legitimate `/* eslint-disable no-unused-vars */` directives appear "unused" under `--report-unused-disable-directives`.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`index.js`:

```js
/* eslint-disable no-unused-vars */
window.addEventListener('logModel', function (_) {
	doSomething()
})
```

`.oxlintrc.json`:

```json
{
	"plugins": [],
	"categories": {"correctness": "off"},
	"rules": {"no-unused-vars": "error"}
}
```

Run:

```
oxlint -c .oxlintrc.json --disable-nested-config --report-unused-disable-directives index.js
```

## Expected

The trailing parameter `_` is unused. With the default `no-unused-vars` options (no `argsIgnorePattern`, `args` defaulting to `after-used`), ESLint flags it:

- ESLint 7.23.0: `'_' is defined but never used.` (1:47)
- ESLint 9.39.4 (core rule, flat config, no plugins): identical message at the same location.

So the `eslint-disable` directive is genuinely needed, and ESLint treats it as used (no unused-directive report).

## Actual

oxlint 1.78.0 reports zero `no-unused-vars` findings and instead emits:

```
index.js:1:1: warning: Unused eslint-disable directive (no problems were reported).
```

The same file without the directive is also silent — `_` is never flagged. A control with a named unused parameter (`function (evt)`) IS flagged, and its diagnostic reads "Unused parameters should start with a '_'." — direct in-product confirmation that the underscore-ignore convention is hardcoded into the native rule's defaults rather than opt-in via `argsIgnorePattern`.

This (a) misses real unused-variable bugs and (b) causes `--report-unused-disable-directives` to wrongly flag valid directives as unused, which is especially damaging during a migration that relies on directive accuracy to sweep suppressions.

## Cross-checked

| Linter | `_` param reported (no directive) | Directive treated as used |
| --- | --- | --- |
| ESLint 7.23.0 (core rule) | yes — `'_' is defined but never used` (1:47) | yes |
| ESLint 9.39.4 (core rule, flat config) | yes — identical | yes |
| oxlint 1.78.0 (native rule) | no | no — directive flagged as unused |
